### PR TITLE
CI-1488: Upgrade draupnir to use pg-14

### PR DESCRIPTION
**Pull request by DragosDumitrache**
Actual footage of this work

This started out as a bit of a red herring, trying to get the vagrant file working. Ended up not doing so, and focused on getting the integration tests working locally.

A quick rundown of what's changed:
- updated Ruby (3.1.2) and Go (1.17) versions to ones that support M1s
- updated Ubuntu to 22.04
- updated the gem locks and the version of pq such that draupnir can interact with the newer postgres
- updated postgres from 11 to 14 in the dockerfile and the corresponding scripts, i.e. `provision.sh` and all the draupnir scripts
- remove mention of `deb` from the readme, and `deb` files, as we have transitioned to `go mod` in 2020.
- With @Nabil372's help, updated the `db.tar` following the same instructions from [here](https://github.com/gocardless/draupnir/commit/e87ffe7fb8603d195236d63835876437b9788a64)

### CODE REVIEW

**Comment by lawrencejones** [comment]
I've been enjoying these PRs, thanks for the entertainment!

**Comment by DragosDumitrache** [comment]
@lawrencejones feel free to join our next treasure hunt, should be an in-person one :trollface:

File: .circleci/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Add any extra build dependencies we need for running draupnir tests
-FROM circleci/golang:1.14
+FROM circleci/golang:1.17
```
Comment (JoeSouthan) [inline .circleci/Dockerfile:2]: Can we swap to the new `cimg/go` as these ones are now deprecated?

**Review by JoeSouthan — COMMENTED** [review]
Seems good, just one thing

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
 
   rubocop:
     docker:
-      - image: circleci/ruby:2.7.1
```
Comment (kevinrobayna) [inline .circleci/config.yml:40]: why not `image: cimg/ruby:3.1.2`?

**Review by kevinrobayna — APPROVED** [review]
LGTM We did something similar on frontier and payments-svc. Knowing that vagrant files and docker files are almost a copy of each other i think we can merge this!

**Comment by nlopes** [comment]
Big fan of draupnir. Would love to see this merged! 👀

File: vagrant/provision.sh [hunk 3]
```diff
@@ -31,9 +31,16 @@ curl -Ss https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
...
+wget https://dl.google.com/go/go1.17.linux-amd64.tar.gz
+sudo tar -xvf go1.17.linux-amd64.tar.gz
+sudo mv go /usr/local
+
+export GOROOT=/usr/local/go
+export GOPATH=$HOME/go
+export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
 
 # install sql-migrate
 go get -v github.com/rubenv/sql-migrate/...
```
Comment (benwh) [inline vagrant/provision.sh:46]: Unchanged, but this line no longer works for me, which breaks vagrant provisioning 😢 


```
root@ubuntu-bionic:/# go get -v github.com/rubenv/sql-migrate/...
github.com/go-gorp/gorp (download)
package github.com/go-gorp/gorp/v3: cannot find package "github.com/go-gorp/gorp/v3" in any of:
        /usr/lib/go-1.10/src/github.com/go-gorp/gorp/v3 (from $GOROOT)
        /root/go/src/github.com/go-gorp/gorp/v3 (from $GOPATH)
package crypto/ed25519: unrecognized import path "crypto/ed25519" (import path does not begin with hostname)
```

File: Dockerfile [hunk 4]
```diff
@@ -1,21 +1,22 @@
-FROM ubuntu:18.04
+FROM ubuntu:22.04
```
Comment (benwh) [inline Dockerfile:1]: If we're updating to 22.04 here, we should probably be doing so in the `Vagrantfile` too

File: vagrant/provision.sh (hunk 3 above)
Comment (benwh) [reply vagrant/provision.sh:46]: Actually please ignore this - I was on a bad commit, this was caused by an old go version in the machine

**Review by benwh — APPROVED** [review]
LGTM - works locally so happy to get this merged in

**Pull request merged**